src/storage/mvcc/txn.rs

```diff
@@ -785,6 +809,23 @@ mod tests {
         must_get_rc(&engine, key, 20, v1);
     }
 
+    #[test]
+    fn test_collapse_prev_rollback() {
+        let engine = engine::new_local_engine(TEMP_DIR, ALL_CFS).unwrap();
+        let (key, value) = (b"key", b"value");
+        let start_ts1 = 1;
+        let start_ts2 = 2;
+
+        must_prewrite_put(&engine, key, value, key, start_ts1);
+        must_rollback(&engine, key, start_ts1);
+        must_get_rollback_ts(&engine, key, start_ts1);
+
+        must_prewrite_put(&engine, key, value, key, start_ts2);
+        must_rollback(&engine, key, start_ts2);
+        must_get_rollback_ts(&engine, key, start_ts2);
+        must_get_rollback_ts_none(&engine, key, start_ts1);
```

Comment:
Maybe we could check the value with ` must_get_none();` at last to ensure the accuracy?